nextcord/role.py

```diff
@@ -449,15 +450,9 @@ async def edit(
             payload['mentionable'] = mentionable
 
         if icon is not MISSING:
-            if icon is None:
-                payload['icon'] = icon
-            elif isinstance(icon, str):
+            if icon is None or isinstance(icon, str):
                 payload['unicode_emoji'] = icon
-                payload['icon'] = None
-            elif isinstance(icon, File):
-                payload['icon'] = _bytes_to_base64_data(icon.fp.read())
-            else:
-                payload['icon'] = _bytes_to_base64_data(icon)
+            payload['icon'] = await _obj_to_base64_data(icon)
```

Comment:
This should probably be an else, since we don't want to set `icon` if it's a unicode emoji?

(Or possibly `icon` could be `None` if it's unicode?)

```suggestion
            else:
                payload['icon'] = await _obj_to_base64_data(icon)
```